### DISCUSSION TITLE
Pi.Alert time

### DISCUSSION THREAD

**Discussion by tesla-76** [tteck/Proxmox#3082] [Help Needed] [answered]
More of a curiosity than a problem, has anyone encountered a problem with Pi.Alert time?
I can't get it to use the time zone for my area, it stays fixed on UTC time.

**Comment by tteck** [comment]
Try executing `dpkg-reconfigure tzdata` in the LXC console and make sure everything is set correctly.

**Reply by tesla-76** [reply]
Thanks but I already tried that command but it doesn't change the time in Pi.Alert. I tried various timezones but nothing, whether Europe or America the time displayed is only UTC time not local.
It had happened as well when I had manually installed Pi.Alert but with "dpkg-reconfigure tzdata" I had in fact solved it. I don't know where the problem lurks. I specify that the program works perfectly, except for indicated time.

**Reply by leiweibau** [reply]
In the Help/FAQ section/page, the 1st entry gives instructions on what to do.
In my case
```
On your PC the following time zone is set for the PHP environment: Europe/Berlin
If this is not the timezone you are in, you should change the timezone in the PHP configuration file. You can find it in this directory: /etc/php/8.2/fpm/php.ini
Search in this file for the entry "date.timezone", remove the leading ";" if necessary and enter the desired timezone. A list with the supported timezones can be found here ([Link](https://www.php.net/manual/en/timezones.php))
```

The path to php.ini is dynamic and depends on the currently used configuration, so it may differ from the path I specified.

**Reply by tesla-76** [reply]
> In the Help/FAQ section/page, the 1st entry gives instructions on what to do. In my case
> 
> ```
> On your PC the following time zone is set for the PHP environment: Europe/Berlin
> If this is not the timezone you are in, you should change the timezone in the PHP configuration file. You can find it in this directory: /etc/php/8.2/fpm/php.ini
> Search in this file for the entry "date.timezone", remove the leading ";" if necessary and enter the desired timezone. A list with the supported timezones can be found here ([Link](https://www.php.net/manual/en/timezones.php))
> ```
> 
> The path to php.ini is dynamic and depends on the currently used configuration, so it may differ from the path I specified.

Thanks for the support, over the weekend I will try to find php.ini. I use your fork of pi.alert with pleasure.

**Reply by tesla-76** [reply, marked as answer]
Then I was able to take a quick look and as indicated by @leiweibau I found the php.ini file in the path indicated, uncommented the string "**date.timezone**" and entered the right TZ value (Europe/Rome for me).
I restarted the LXC container all happy to have a real clock and not Easter Island time, which may be a nice place but is useless to me, and... surprise 🎉 it was not working! 🤔
Before I despaired thinking that an ancient time demon had taken over my home server or pretending to have a the server on Easter Island, starting with the suggestion to edit the **php.ini** file I searched for more information and came to the conclusion that additional php.ini files located in the other directories under "**/etc/php/8.2/**" needed to be edited. Rebooted and now the clock is perfect.

Thanks for the help.

**Reply by Evil-Teddy** [reply]
![pialert](https://github.com/user-attachments/assets/f83e8e69-9d8c-4420-8c69-1234b9885b58)

I uncommented all "date.timezone" in all "php.ini"s in  "/etc/php/8.2/" and I have still the wrong time whon in my pi.alert.
See the shown time in the screenshot and its 12:40 now.

**Reply by tteck** [reply]
This one `/etc/php/8.2/fpm/php.ini`?

**Reply by leiweibau** [reply]
The help/FAQ page will show you the exact path to php.ini and what you have to do. It is not enough just to uncomment, but also to set the correct time zone. A link to the possible time zones can also be found in the help section.

**Reply by Evil-Teddy** [reply]
> This one `/etc/php/8.2/fpm/php.ini`?

yes this



> The help/FAQ page will show you the exact path to php.ini and what you have to do. It is not enough just to uncomment, but also to set the correct time zone. A link to the possible time zones can also be found in the help section.

I uncomment and set it to Europe/Berlin

**Reply by leiweibau** [reply]
On the Help/FAQ page, do you also get: "On your PC the following time zone is set for the PHP environment: Europe/Berlin"?

**Reply by Evil-Teddy** [reply]
The PC where pi.alert is running on?

**Reply by leiweibau** [reply]
> The PC where pi.alert is running on?

Via the web frontend of Pi.Alert -> Help/FAQ -> General -> First entry

<img width="713" alt="screen" src="https://github.com/user-attachments/assets/c15180f2-c79b-4ef0-9bc0-88b9959a7892">

**Reply by Evil-Teddy** [reply]
![pialert-aktuell](https://github.com/user-attachments/assets/308656ae-1b93-408b-8e2b-cd658cbd59aa)

But I already set the timezone in /etc/php/8.2/apache2/php.ini to Europe/berlin

**Reply by leiweibau** [reply]
To be on the safe side, check the configuration again. Even if you have probably already done so, restart the system/VM again. Unfortunately, I'm short on time at the moment as I'm just taking my lunch break. If in doubt, I can get back to you later in the day.

**Reply by Evil-Teddy** [reply]
Okay I will restart and check everything again and I write here again then

**Reply by Evil-Teddy** [reply]
Okay after checking everything and restart it seems to work now. I have the correct time now. Thank you for your help :-)
